src/prefect/events/filters.py

```diff
@@ -0,0 +1,213 @@
+from typing import List, Optional, Tuple, cast
+from uuid import UUID
+
+import pendulum
+
+from prefect.events.schemas import Event, Resource, ResourceSpecification
+from prefect.server.utilities.schemas import DateTimeTZ, PrefectBaseModel
+
+try:
```

Comment:
nit: I'm wondering if we have these imports under the `HAS_PYDANTIC_V2` it might be 1% easier to search and update our imports whenever we're on pydantic v2 only.